v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -1171,6 +1172,103 @@ func assignMapFromMap(
 	}, nil
 }
 
+// assignUserAssignedIdentityMapFromArray will generate a code fragment to populate a userAssignedIdentity array from
+// a map whose key is the ARM ID of the userAssignedIdentity
+// <arr> := make([]<arrType>, 0, len(source.UserAssignedIdentities))
+//
+//	for key, _ := range source.UserAssignedIdentities {
+//	   ref := genruntime.CreateResourceReferenceFromARMID(key)
+//	   <arr> = append(<arr>, UserAssignedIdentityDetails{Reference: ref})
+//	}
+//
+// <writer> = <arr>
+func assignUserAssignedIdentityMapFromArray(
+	sourceEndpoint *TypedConversionEndpoint,
+	destinationEndpoint *TypedConversionEndpoint,
+	conversionContext *PropertyConversionContext) (PropertyConversion, error) {
+
+	// Require both source and destination to not be bag items
+	if sourceEndpoint.IsBagItem() || destinationEndpoint.IsBagItem() {
+		return nil, nil
+	}
+
+	// Require source to be a map type
+	_, sourceIsMap := astmodel.AsMapType(sourceEndpoint.Type())
+	if !sourceIsMap {
+		return nil, nil
+	}
+
+	// Require destination to be an array type
+	destinationArray, destinationIsArray := astmodel.AsArrayType(destinationEndpoint.Type())
+	if !destinationIsArray {
+		return nil, nil
+	}
+
+	if sourceEndpoint.Name() != astmodel.UserAssignedIdentitiesProperty {
+		return nil, nil
+	}
+
+	if destinationEndpoint.Name() != astmodel.UserAssignedIdentitiesProperty {
+		return nil, nil
+	}
+
+	destinationElement := destinationArray.Element()
+
+	// TODO: Need an outer nil-check
+	return func(reader dst.Expr, writer func(dst.Expr) []dst.Stmt, knownLocals *astmodel.KnownLocalsSet, generationContext *astmodel.CodeGenerationContext) []dst.Stmt {
+
+		// <source>List := make([]<type>, 0, len(<source>)
+		tempId := knownLocals.CreateSingularLocal(sourceEndpoint.Name(), "List")
+		declaration := astbuilder.ShortDeclaration(
+			tempId,
+			astbuilder.MakeAppendSlice(destinationArray.AsType(generationContext), astbuilder.CallFunc("len", reader)))
+
+		loopLocals := knownLocals.Clone()
+		keyID := loopLocals.CreateLocal(sourceEndpoint.Name(), "Key")
+
+		intermediateDestination := loopLocals.CreateLocal(destinationEndpoint.Name(), "Ref")
+
+		conversion, err := CreateTypeConversion(
```

Comment:
@theunrepentantgeek - other callers of CreateTypeConversion do it above, rather than in this function, but it wasnt' totally clear to me how to pull that off and use `knownLocals` to pick an intermediate destination name.

Let me know if there's a different way this should be being done.

Reply:
Definitely should be doing all the creation of conversions in the outer method - the returned func should never panic, it should be doing code generation that's known to succeed.

Reply:
Do you have any suggestions on how exactly I should go about doing that in this case, where the `intermediateDestination` of the conversion is to a local variable whose name is determined by `loopLocals.CreateLocal`?

I don't have access to the `knownLocals` outside of the context of this function.

Maybe I'm confused and the "Name" aspect of the conversion endpoint doesn't matter as much as I thought it did because the function returned (the actual conversion function) takes in the name to assign to? Looking at `assignMapFromMap` that must be the case...

I'll move this up

Reply:
Moved